tools/checkpoint_conversion/convert_gpt2_checkpoints.py

```diff
@@ -0,0 +1,312 @@
+        keras_nlp_model.get_layer(
+            f"transformer_layer_{i}"
+        )._feedforward_layernorm.beta.assign(weights[f"model/h{i}/ln_2/b"])
+
+    keras_nlp_model.get_layer("layer_norm").gamma.assign(
+        weights["model/ln_f/g"]
+    )
+
+    keras_nlp_model.get_layer("layer_norm").beta.assign(weights["model/ln_f/b"])
+
+    # Save the model.
+    print(f"\n-> Save KerasNLP model weights to `{preset}.h5`.")
+    keras_nlp_model.save_weights(f"{preset}.h5")
+
+    return keras_nlp_model
+
+
+def define_tokenizer(
+    preset, num_params, hf_model_name, check_cloud_output=False
+):
+    print("\n-> Define the tokenizers.")
+    extract_dir = extract_dir = EXTRACT_DIR.format(num_params)
+    merges_path = os.path.join(extract_dir, "vocab.bpe")
+    vocab_path = os.path.join(extract_dir, "encoder.json")
+
+    keras_nlp_tokenizer = keras_nlp.models.GPT2Tokenizer(
+        vocabulary=vocab_path,
+        merges=merges_path,
+    )
+    hf_tokenizer = transformers.AutoTokenizer.from_pretrained(hf_model_name)
+
+    if not check_cloud_output:
+        print("\n-> Print MD5 checksum of the vocab files.")
+        print(f"`{vocab_path}` md5sum: ", get_md5_checksum(vocab_path))
+        print(f"`{merges_path}` md5sum: ", get_md5_checksum(merges_path))
+
+    return keras_nlp_tokenizer, hf_tokenizer
+
+
+def check_output(
+    preset,
+    keras_nlp_model,
+    keras_nlp_tokenizer,
+    hf_model,
+    hf_tokenizer,
+    check_cloud_output=False,
+):
+    print("\n-> Check the outputs.")
+    input_str = ["the quick brown fox ran, galloped and jumped."]
+
+    # KerasNLP
+    token_ids = keras_nlp_tokenizer(input_str)
+    padding_mask = token_ids != 0
+
+    keras_nlp_inputs = {
+        "token_ids": token_ids.to_tensor(),
+        "padding_mask": padding_mask.to_tensor(),
+    }
+    keras_nlp_output = keras_nlp_model.predict(keras_nlp_inputs)
+
+    # HF
+    hf_inputs = hf_tokenizer(input_str, return_tensors="pt")
+    hf_output = hf_model(**hf_inputs).last_hidden_state
+
+    print("KerasNLP output:", keras_nlp_output[0, 0, :10])
+    print("HF output:", hf_output[0, 0, :10])
+    print("Difference:", np.mean(keras_nlp_output - hf_output.detach().numpy()))
+
+    if check_cloud_output:
+        print("\n-> Check KerasNLP cloud output.")
+        keras_nlp_cloud_model = keras_nlp.models.GPT2.from_preset(
+            preset,
+            load_weights=True,
+        )
+        keras_nlp_cloud_output = keras_nlp_cloud_model.predict(keras_nlp_inputs)
+        print(
+            "Difference:",
+            tf.reduce_mean(keras_nlp_output - keras_nlp_cloud_output),
+        )
+    else:
+        # Show the MD5 checksum of the model weights.
+        print("Model md5sum: ", get_md5_checksum(f"./{preset}.h5"))
+
+    return keras_nlp_output
+
+
+def main(preset, check_cloud_output):
+    num_params = PRESET_MAP[preset][0]
+    hf_model_name = PRESET_MAP[preset][1]
+
+    download_model(preset, num_params)
```

Comment:
It seems slightly weird to me that we download directly from gpt, but then use huggingface to verify. Is this the general practice we want?

The original gpt-2 code is not pip friendly, it is not a package and there is no `setup.py`. Seems like we have a few options here:
1) Use huggingface as both the weights source and how we verify output.
2) Download the original weights but verify using huggingface (and assume the outputs line up).
3) Download the original weights and use the original code to verify output.

For 3) we would need to provide some instructions on cloning the original repo so we can use it. Or just `os.system(f"git clone {repo_url}")` if we really want this to be plug and play.

@jbischof @abheesht17 curious thoughts. This is "best practice" we will need to decide that we will probably hit a lot.

Reply:
I am fine using HF as a SoT for now. Notebooks are highly optimized for this kind of "one and done" analysis, so if this gets too complicated we can go back Colab.